### PR TITLE
DEV-1602 Added classes and translations for payment complements

### PR DESCRIPTION
**Pull request by juanmex**
Input:
```
{
  "receipt": {
    "date": "2021-12-23T12:09:50",
    "xsi:schemaLocation": "http://www.sat.gob.mx/cfd/4 http://www.sat.gob.mx/sitio_internet/cfd/4/cfdv40.xsd http://www.sat.gob.mx/Pagos http://www.sat.gob.mx/sitio_internet/cfd/Pagos/Pagos20.xsd",
    "export_code": "01",
    "version": "4.0",
    //"payment_way": "03",
    "subtotal": "0",
    "currency": "MXN",
    "total": "0",
    "type_of_receipt": "P",
    //"payment_method": "PUE",
    "expedition_place": "27010",
    "issuer_attributes": {
      "rfc": "EKU9003173C9",
      "name": "ESCUELA KEMPER URGATE SA DE CV",
      "fiscal_regime": 601
    },
    "receiver_attributes": {
      "rfc": "AAA010101AAA",
      "name": "ACCEM SERVICIOS EMPRESARIALES SC",
      "zip_code": "97000",
      "fiscal_regime": 601,
      "cfdi_usage": "CP01"
    },
    "concepts_attributes": [
      {
        "taxable_code": "01",
        "prod_or_svc_key": "84111506",
        "identification_number": 1,
        "quantity": "1",
        "unit_measure_key": "ACT",
        "unit_measure": "ACT",
        "description": "Pago",
        "unit_value": "0",
        "amount": "0"
      }
    ],
    "payment_attributes":{
      "payment_date": "2021-12-01T08:00:00",
      "payment_form": "01",
      "currency":"MXN",
      "exchange_rate":"1",
      "amount":"15680.00",
      "bank_name":"SANTANDER",
      "totals": {
        "total_retained_iva": "0.00",
        "total_retained_isr": "0.00",
        "total_retained_ieps": "0.00",
        "total_transferred_base_iva_16": "15680.00",
        "total_transferred_tax_iva_16": "2508.80",
        "total_transferred_base_iva_8": "0.00",
        "total_transferred_tax_iva_8": "0.00",
        "total_transferred_base_iva_0": "0.00",
        "total_transferred_tax_iva_0": "0.00",
        "total_transferred_base_iva_exempt": "0.00",
        "total_payment_amount": "15680.00"
      },
      "related_docs": [
        {
          "uuid":"63357C27-9698-49F5-BE69-913E9E7E9CF8",
          "serie":"ASD",
          "folio":"23",
          "currency":"MXN",
          "payment_number":"1",
          "exchange_rate":"1",
          "previous_debt":"25000.00",
          "paid_amount":"15680.00",
          "new_debt":"9320.00",
          "taxable_code":"02",
          "taxes": {
            "transfers": [
              {
                "base": "15680.00",
                "tax":"002",
                "factor_type":"Tasa",
                "rate_or_fee":"0.160000",
                "import":"2508.80"
              }
            ],
            "retentions": [
              {
                "base": "10000.00",
                "tax":"002",
                "factor_type":"Tasa",
                "rate_or_fee":"0.1600",
                "import":"100.00"
              }
            ]
          }
        }
      ]
    }
  }
}
```

Output:
```
<cfdi:Comprobante
	xmlns:cfdi="http://www.sat.gob.mx/cfd/4"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xmlns:pago20="http://www.sat.gob.mx/Pagos20" Version="4.0" Fecha="2021-12-23T12:09:50" SubTotal="0" Moneda="MXN" Total="0" TipoDeComprobante="P" LugarExpedicion="27010" Exportacion="01" xsi:schemaLocation="http://www.sat.gob.mx/cfd/4 http://www.sat.gob.mx/sitio_internet/cfd/4/cfdv40.xsd">
	<cfdi:Emisor Rfc="EKU9003173C9" Nombre="ESCUELA KEMPER URGATE SA DE CV" RegimenFiscal="601"/>
	<cfdi:Receptor Rfc="AAA010101AAA" Nombre="ACCEM SERVICIOS EMPRESARIALES SC" UsoCFDI="CP01" DomicilioFiscalReceptor="97000" RegimenFiscalReceptor="601"/>
	<cfdi:Conceptos>
		<cfdi:Concepto ClaveProdServ="84111506" NoIdentificacion="1" Cantidad="1" ClaveUnidad="ACT" Unidad="ACT" Descripcion="Pago" ValorUnitario="0" Importe="0" ObjetoImp="01"/>
	</cfdi:Conceptos>
	<pago20:Pagos Version="2.0">
		<pago20:Totales TotalRetencionesIVA="0.00" TotalRetencionesISR="0.00" TotalRetencionesIEPS="0.00" TotalTrasladosBaseIVA16="15680.00" TotalTrasladosImpuestoIVA16="2508.80" TotalTrasladosBaseIVA8="0.00" TotalTrasladosImpuestoIVA8="0.00" TotalTrasladosBaseIVA0="0.00" TotalTrasladosImpuestoIVA0="0.00" TotalTrasladosBaseIVAExento="0.00" MontoTotalPagos="15680.00"/>
		<pago20:Pago FechaPago="2021-12-01T08:00:00" FormaDePagoP="01" MonedaP="MXN" TipoCambioP="1" Monto="15680.00" NomBancoOrdExt="SANTANDER">
			<pago20:DoctoRelacionado IdDocumento="63357C27-9698-49F5-BE69-913E9E7E9CF8" Serie="ASD" Folio="23" MonedaDR="MXN" NumParcialidad="1" EquivalenciaDR="1" ImpSaldoAnt="25000.00" ImpPagado="15680.00" ImpSaldoInsoluto="9320.00" ObjetoImpDR="02">
				<pago20:ImpuestosDR>
					<pago20:TrasladosDR>
						<pago20:TrasladoDR BaseDR="15680.00" ImpuestoDR="002" TipoFactorDR="Tasa" TasaOCuotaDR="0.160000" ImporteDR="2508.80"/>
					</pago20:TrasladosDR>
					<pago20:RetencionesDR>
						<pago20:RetencionDR BaseDR="10000.00" ImpuestoDR="002" TipoFactorDR="Tasa" TasaOCuotaDR="0.1600" ImporteDR="100.00"/>
					</pago20:RetencionesDR>
				</pago20:ImpuestosDR>
			</pago20:DoctoRelacionado>
		</pago20:Pago>
	</pago20:Pagos>
</cfdi:Comprobante>
```

### CODE REVIEW

File: lib/buho_cfdi/params_builder.rb [hunk 1]
```diff
@@ -130,5 +131,32 @@ def build_taxes
...
+
+    def build_payment_info
+      if params.include?(:receipt) && params.fetch(:receipt).include?(:payment_attributes)
+        @receipt.build_child! ::Nodes::PaymentInfo, params.fetch(:receipt).fetch(:payment_attributes)
+        @receipt.nodes_paymentinfo.build_child! ::Nodes::PaymentTotals, params.fetch(:receipt).fetch(:payment_attributes).fetch(:totals)
+        @receipt.nodes_paymentinfo.build_children ::Nodes::RelatedDoc
+
+          params.fetch(:receipt).fetch(:payment_attributes).fetch(:related_docs).each do |params|
+            related_doc = @receipt.nodes_paymentinfo.nodes_relateddoc.add params
+            if !params[:taxes].blank? && !params[:taxes][:transfers].blank?
```
Comment (yusent) [inline lib/buho_cfdi/params_builder.rb]:
```suggestion
            if params.dig(:taxes, :transfers).present?
```

File: lib/buho_cfdi/params_builder.rb [hunk 2]
```diff
@@ -130,5 +131,32 @@ def build_taxes
...
+              related_doc.build_children ::Nodes::RelatedDocTransfer
+              params[:taxes][:transfers].each do |transfer|
+                related_doc.nodes_relateddoctransfer.add transfer
+              end
+            end
+
+            if !params[:taxes].blank? && !params[:taxes][:retentions].blank?
```
Comment (yusent) [inline lib/buho_cfdi/params_builder.rb]:
```suggestion
            if params.dig(:taxes, :retentions).present?
```

File: lib/buho_cfdi/params_builder.rb (hunk 1 above)
Comment (juanmex) [reply lib/buho_cfdi/params_builder.rb]: Solved with dcef64a371831a080036fd1bdf170a6cb88b86e9

File: lib/buho_cfdi/params_builder.rb (hunk 2 above)
Comment (juanmex) [reply lib/buho_cfdi/params_builder.rb]: Solver with dcef64a371831a080036fd1bdf170a6cb88b86e9

**Review by yusent — APPROVED** [review]

**Review by ococom1476 — APPROVED** [review]

**Review by ococom1476 — APPROVED** [review]

**Pull request merged**